processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -56,19 +63,62 @@
   public SelectorDimFilter(
       @JsonProperty("dimension") String dimension,
       @JsonProperty("value") String value,
-      @JsonProperty("extractionFn") ExtractionFn extractionFn
+      @JsonProperty("extractionFn") ExtractionFn extractionFn,
+      @JsonProperty("dimensions") String[] dimensions
   )
   {
-    Preconditions.checkArgument(dimension != null, "dimension must not be null");
+    if (dimension == null && dimensions == null) {
+      throw new IAE("dimension or dimensions must not be null");
+    }
+    if (dimensions != null) {
+      Preconditions.checkArgument(dimension == null, "either dimension or dimensions must be null");
+      Preconditions.checkArgument(value == null, "value must be null when dimensions is used");
+      Preconditions.checkArgument(dimensions.length >= 2, "dimensions must have a least 2 dimensions");
+    }
 
     this.dimension = dimension;
     this.value = Strings.nullToEmpty(value);
     this.extractionFn = extractionFn;
+    this.dimensions = dimensions;
+  }
+
+  public SelectorDimFilter(
+      String dimension,
+      String value,
+      ExtractionFn extractionFn
+  )
+  {
+    this(dimension, value, extractionFn, null);
   }
 
   @Override
   public byte[] getCacheKey()
   {
+    if (dimensions != null) {
```

Comment:
This looks good, but it's desired to use ```CacheKeyBuilder```. It will be much easier to use and understand.